posts/2022-08-15-openliberty-microprofile-configuration.adoc

```diff
@@ -0,0 +1,252 @@
+
+MicroProfile and Open Liberty have standardized the use of these parameter files with additional properties like ordianls. Different type of files where the constants and run time variables can be hosted are 
+
+             server.xml
+             Environment based files/variables in source code
+             microprofile-config.properties
+             bootstrap.properties
+             CustomConfig properties (anything which implement org.eclipse.microprofile.config.spi.ConfigSource)
+
+All these property files contains an ordinal value to set the reading order of the file by the application. 
+
+Applications need to be hosted on multiple environments like development, test and production. There will be several variables whose values would be changing across these environments and containerized platforms has different configuration methods like given below to inject the variable values on run time to the application.
+
+            Configuring through environment variables
+            Configuring through ConfigMap
+            Configuring through Secrets
+            Configuring through configuration file in a volume
+
+When values of variables configured on environment variable, configmap and secrets are changed, application pod will be restarted to complete the injection, but that's not the case with configuration through volume. 
+
+Architecture diagram of Open Liberty config files flow from GitHub through a build tool to Kubernetes is as given below
+
+image::/img/blog/ol-configuration1.png[Open Liberty configuration file flow,width=100%,align="left"]
+
+== Different data injection methods in Kubernetes/OpenShift
+
+There are multiple ways to inject data into application containers. This blog will discuss those methods one by one. The commands given in this blogs are OpenShift commands, but you can run the same on Kubernetes by replacing 'oc' by 'kubeclt'
+
+For the purpose of demonstration, i have defined variables in below given files and the values for the same is injected from different container data sources.
+
+server.xml&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;microprofile-config.properties
```

Comment:
This can't be the best way to format these headings

Reply:
Will check better option later